File: POLIMORFISMO/bin/Videojuego.py
class Videojuego:
    def __init__(self, nombre, plataforma, cantidadJugadores=1):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidadJugadores = cantidadJugadores

    def mostrar(self):
        print(f"Nombre: {self.nombre}, Plataforma: {self.plataforma}, Jugadores: {self.cantidadJugadores}")

    def agregarJugadores(self):
        """Agrega 1 jugador por defecto."""
        self.cantidadJugadores += 1

    def agregarJugadoresConCantidad(self, cantidad):
        """Agrega una cantidad específica de jugadores."""
        self.cantidadJugadores += cantidad


juego1 = Videojuego("FIFA 22", "PlayStation", 4)
juego2 = Videojuego("Minecraft", "PC")


juego1.mostrar()
juego2.mostrar()


juego1.agregarJugadores()
juego2.agregarJugadoresConCantidad(3)


juego1.mostrar()
juego2.mostrar()

File: POLIMORFISMO/EJERCICIO-1/Videojuego.java
public class Videojuego{
    String nombre;
    String plataforma;
    int cantidaddejugadores;

    public Videojuego(String nombre, String plataforma, int cantidaddejugadores){
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidaddejugadores = cantidaddejugadores;
    }
    public Videojuego(){
        this.nombre = "Desconocido";
        this.plataforma = "PC";
        this.cantidaddejugadores = 1;
    }
    public String mostrar(){
        return "Nombre: "+nombre+" Plataforma: "+plataforma+" Cantidad: "+cantidaddejugadores;
    }
    public int cantidaddeJugadores(){
        this.cantidaddejugadores = cantidaddejugadores + 1;
        return cantidaddejugadores;
    }
    public int cantidaddeJugadores(int cantidad){
        this.cantidaddejugadores = cantidaddejugadores + cantidad;
        return cantidaddejugadores;
    }

    public static void main(String[] args) {
        Videojuego juego1 = new Videojuego("FIFA 22", "PlayStation", 4);
        Videojuego juego2 = new Videojuego();

  
    System.out.println(juego1.mostrar());
    System.out.println(juego2.mostrar());

   
    juego1.cantidaddeJugadores(); 
    juego2.cantidaddeJugadores(3); 

    System.out.println(juego1.mostrar());
    System.out.println(juego2.mostrar());
        
    }
    
}